Framework/DataHandling/src/SaveNXcanSAS.cpp

```diff
@@ -393,48 +393,51 @@ void addData1D(H5::Group &data, Mantid::API::MatrixWorkspace_sptr workspace) {
 
   if (workspace->isHistogramData()) {
     std::vector<double> qValueCentres;
-    Mantid::Kernel::VectorHelper::convertToBinCentre(qValue, qValueCentres);
+    Mantid::Kernel::VectorHelper::convertToBinCentre(qValue.rawData(),
+                                                     qValueCentres);
     writeArray1DWithStrAttributes(data, sasDataQ, qValueCentres, qAttributes);
   } else {
-    writeArray1DWithStrAttributes(data, sasDataQ, qValue, qAttributes);
+    writeArray1DWithStrAttributes(data, sasDataQ, qValue.rawData(),
+                                  qAttributes);
   }
 
   //-----------------------------------------
   // Add I with units + uncertainty definition
-  const auto intensity = workspace->readY(0);
+  const auto &intensity = workspace->y(0);
   std::map<std::string, std::string> iAttributes;
   auto iUnit = getIntensityUnit(workspace);
   iUnit = getIntensityUnitLabel(iUnit);
   iAttributes.emplace(sasUnitAttr, iUnit);
   iAttributes.emplace(sasUncertaintyAttr, sasDataIdev);
 
-  writeArray1DWithStrAttributes(data, sasDataI, intensity, iAttributes);
+  writeArray1DWithStrAttributes(data, sasDataI, intensity.rawData(),
+                                iAttributes);
 
   //-----------------------------------------
   // Add Idev with units
-  const auto intensityUncertainty = workspace->readE(0);
+  const auto &intensityUncertainty = workspace->e(0);
   std::map<std::string, std::string> eAttributes;
   eAttributes.insert(
       std::make_pair(sasUnitAttr, iUnit)); // same units as intensity
 
-  writeArray1DWithStrAttributes(data, sasDataIdev, intensityUncertainty,
-                                eAttributes);
+  writeArray1DWithStrAttributes(data, sasDataIdev,
+                                intensityUncertainty.rawData(), eAttributes);
 
   //-----------------------------------------
   // Add Qdev with units if available
   if (workspace->hasDx(0)) {
-    const auto qResolution = workspace->readDx(0);
+    const auto qResolution = workspace->dx(0);
```

Comment:
X uncertainties are always for points (Dx represents a delta-Q for bins, not an uncertainty for bin edges), so I think the branching below should be removed. But just use `pointStandardDeviations(0)` here to be on the safe side.